cli/src/main/scala/com/ibm/sparktc/sparkbench/datageneration/GraphDataGen.scala

```diff
@@ -0,0 +1,118 @@
+/**
+  * (C) Copyright IBM Corp. 2015 - 2017
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  *
+  */
+package com.ibm.sparktc.sparkbench.datageneration
+
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+import com.ibm.sparktc.sparkbench.utils.SparkBenchException
+import com.ibm.sparktc.sparkbench.utils.GeneralFunctions.{getOrDefault, getOrThrow, time, any2Long}
+import com.ibm.sparktc.sparkbench.workload.{Workload, WorkloadDefaults}
+import org.apache.spark.sql.types.{LongType, StringType, StructField, StructType}
+import org.apache.spark.graphx.util.GraphGenerators
+
+object GraphDataGen extends WorkloadDefaults {
+
+  val name = "graph-data-generator"
+  val mu = 4.0
+  val sigma = 1.3
+  val seed = -1L
+  val numOfPartitions = 0
+
+  override def apply(m: Map[String, Any]): GraphDataGen = {
+      val numVertices = getOrThrow(m, "vertices").asInstanceOf[Int]
+      val muu = getOrDefault[Double](m, "mu", mu)
+      val sigmaa = getOrDefault[Double](m, "sigma", sigma)
+      val numPartitions = getOrDefault[Int](m, "partitions", numOfPartitions)
+      val seedd = getOrDefault[Long](m, "seed", seed, any2Long)
+      val output = {
+        val str = getOrThrow(m, "output").asInstanceOf[String]
+        val s = verifySuitabilityOfOutputFileFormat(str)
+        Some(s)
+      }
+
+    new GraphDataGen(
+      numVertices = numVertices,
+      input = None,
+      output = output,
+      mu = muu,
+      sigma = sigmaa,
+      seed = seedd,
+      numPartitions = numPartitions
```

Comment:
I agree that "defaultX" is way better than weird misspellings. I have changed that. The only reason I used temporary variables here instead of one-lining is because `output` needs to both be extracted from the map and then passed to verifySuitabilityOfWorkload(), and one-lining that would be visually messy, IMO. I'm willing to have my mind changed though :)

Reply:
Yeah, I was imagining one-lining the four getOrDefaults and leaving `output` as it is.  But I'm happy with this.